### PR TITLE
perl*Packages: Fix all packages

### PR DESCRIPTION
**Pull request by dasJ**
This is mostly done by disabling the tests or the entire package on
Darwin

###### Description of changes

ZHF: #172160 cc @NixOS/nixos-release-managers 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 174722` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>perl532Packages.CacheKyotoTycoon</li>
    <li>perl532Packages.Furl</li>
    <li>perl532Packages.HTTPCookieJar</li>
    <li>perl534Packages.CacheKyotoTycoon</li>
    <li>perl534Packages.Furl</li>
    <li>perl534Packages.HTTPCookieJar</li>
  </ul>
</details>

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 174722` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gcstar</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>auto-multiple-choice</li>
    <li>gtk-pipe-viewer</li>
    <li>perl532Packages.BoostGeometryUtils</li>
    <li>perl532Packages.CacheKyotoTycoon</li>
    <li>perl532Packages.Furl</li>
    <li>perl532Packages.GooCanvas2</li>
    <li>perl532Packages.GooCanvas2CairoTypes</li>
    <li>perl532Packages.Gtk3</li>
    <li>perl532Packages.Gtk3SimpleList</li>
    <li>perl532Packages.MathClipper</li>
    <li>perl534Packages.BoostGeometryUtils</li>
    <li>perl534Packages.CacheKyotoTycoon</li>
    <li>perl534Packages.Furl</li>
    <li>perl534Packages.GooCanvas2</li>
    <li>perl534Packages.GooCanvas2CairoTypes</li>
    <li>perl534Packages.Gtk3</li>
    <li>perl534Packages.Gtk3SimpleList</li>
    <li>perl534Packages.MathClipper</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>perl532Packages.CatalystAuthenticationStoreLDAP</li>
    <li>perl532Packages.ExtUtilsCppGuess</li>
    <li>perl532Packages.GlibObjectIntrospection</li>
    <li>perl532Packages.Gnome2Canvas</li>
    <li>perl532Packages.HTTPCookieJar</li>
    <li>perl532Packages.ModuleBuildWithXSpp</li>
    <li>perl534Packages.CatalystAuthenticationStoreLDAP</li>
    <li>perl534Packages.ExtUtilsCppGuess</li>
    <li>perl534Packages.GlibObjectIntrospection</li>
    <li>perl534Packages.Gnome2Canvas</li>
    <li>perl534Packages.HTTPCookieJar</li>
    <li>perl534Packages.ModuleBuildWithXSpp</li>
  </ul>
</details>

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -1619,6 +1619,7 @@ let
       homepage = "https://github.com/mongodb-labs/mongo-perl-bson-xs";
       description = "XS implementation of MongoDB's BSON serialization (EOL)";
       license = lib.licenses.asl20;
+      platforms = lib.platforms.linux; # configure phase fails with "ld: unknown option: -mmacosx-version-min=10.12"
```
Comment (zakame) [inline pkgs/top-level/perl-packages.nix:1622]: Would setting `export LD=$CC` environment work for this?  I reckon that's how we do on linker failures like this one on Darwin.

**Review by Mindavi — APPROVED** [review]

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (dasJ) [reply pkgs/top-level/perl-packages.nix:1622]: Tried it, didn't work

**Pull request merged**